src/isolation/filemap.rs

```diff
@@ -7,27 +7,67 @@ use std::{
 };
 
 use clean_path::clean;
+#[cfg(target_os = "linux")]
+use libc::{O_DIRECT, O_SYNC};
 use tempfile::TempDir;
 use uuid::Uuid;
 
 use crate::isolation::{
 	fd::UhyveFileDescriptorLayer, split_guest_and_host_path, tempdir::create_temp_dir,
 };
 
+/// Defines cache-related behaviors that will be forced upon [`crate::hypercall::open`],
+/// primarily useful for e.g. I/O benchmarking.
+#[cfg(target_os = "linux")]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub enum UhyveIoMode {
+	/// Do not enforce any particular behavior.
+	#[default]
+	Default,
+	/// Append the O_DIRECT flag to bypass the host's page cache.
+	HostDirect,
+	/// Append the O_DIRECT flag to bypass the host's page cache and block until writes are finished on the host.
+	HostDirectSync,
+}
+
+#[cfg(target_os = "linux")]
+impl From<Option<String>> for UhyveIoMode {
+	fn from(s: Option<String>) -> Self {
+		match s.unwrap_or_default().to_lowercase().as_str() {
+			"" | "default" => UhyveIoMode::Default,
+			// TODO: Remove "direct"/"directsync" flags once we can control whether
+			// page caches can (not) be skipped in the guest as well. Currently only
+			// there for convenience reasons.
+			"direct" | "host-direct" | "hostdirect" => UhyveIoMode::HostDirect,
+			"directsync" | "host-directsync" | "hostdirectsync" => UhyveIoMode::HostDirectSync,
```

Comment:
"directsync" sounds a bit ... dense, something like "direct,sync" might be clearer in meaning.

Reply:
wouldn't this be extra weird given that I intend to make the `host-` prefix official?

I guess one could do `host=direct,sync` but then I'd have to do separation so that e.g. `host=sync,direct` works as well. If you think that's the way forward, then /shrug

Reply:
touché